apiserver/src/main/java/org/dependencytrack/resources/v1/AdvisoriesResource.java

```diff
@@ -0,0 +1,380 @@
+        final var publisher = doc.getDocument().getPublisher().getNamespace().toString();
+
+        // Create a new advisory that we have already "seen" and was just fetched
+        final var advisory = new Advisory();
+        advisory.setTitle(doc.getDocument().getTitle());
+        advisory.setUrl(fileName);
+        advisory.setContent(content);
+        advisory.setLastFetched(Instant.now());
+        advisory.setPublisher(publisher);
+        advisory.setName(doc.getDocument().getTracking().getId());
+        advisory.setVersion(doc.getDocument().getTracking().getVersion());
+        advisory.setFormat("CSAF");
+        advisory.setSeen(true);
+
+        // Create a pseudo-source populated by the information we have
+        var manual = new CsafSource();
+        manual.setDomain(true);
+        manual.setUrl(URI.create(publisher).getHost());
+        manual.setName(publisher);
+
+        final var bov = ModelConverter.convert(result, manual);
+        if (bov == null) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Could not extract vulnerability information").build();
+        }
+
+        for (final var v : bov.getVulnerabilitiesList()) {
+            final Vulnerability vuln = BovModelConverter.convert(bov, v, true);
+            final List<VulnerableSoftware> vsList = BovModelConverter.extractVulnerableSoftware(bov);
+
+            LOGGER.debug("Synchronizing vulnerability " + vuln.getVulnId());
+            final Vulnerability persistentVuln = qm.synchronizeVulnerability(vuln, false);
+            qm.synchronizeVulnerableSoftware(persistentVuln, vsList, Vulnerability.Source.CSAF);
+
+            advisory.addVulnerability(persistentVuln);
+        }
+
+        // Sync the document into the database, replacing an older version with the same
+        // combination of tracking ID and publisher namespace if necessary
+        qm.synchronizeAdvisory(advisory);
+
+        return Response.ok("File uploaded successfully: " + fileName).build();
+    }
+
+    @DELETE
+    @Path("/{advisoryId}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Deletes an advisory", description = "<p>Requires permission <strong>VULNERABILITY_ANALYSIS_UPDATE</strong></p>")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "204", description = "Advisory removed successfully"),
+            @ApiResponse(responseCode = "401", description = "Unauthorized"),
+            @ApiResponse(responseCode = "404", description = "The advisory could not be found")
+    })
+    @PermissionRequired(Permissions.Constants.VULNERABILITY_ANALYSIS_UPDATE)
+    public Response deleteAdvisory(@PathParam("advisoryId") String advisoryId) {
+        try (QueryManager qm = new QueryManager()) {
+            final Advisory entity = qm.getObjectById(Advisory.class, advisoryId);
+            if (entity != null) {
+                qm.delete(entity);
```

Comment:
When multiple persistence operations are performed, please wrap them in a `qm.runInTransaction` or `qm.callInTransaction` block to ensure atomicity.